src/parser/flags.ts

```diff
@@ -40,6 +40,34 @@ export const integer = build({
   },
 })
 
+/**
+ * Initializes a string as a URL. Throws an error
+ * if the string is not a valid URL.
+ */
+export const url = build({
+  parse: async input => {
+    try {
+      return new URL(input)
+    } catch {
+      throw new Error(`Expected a valid url but received: ${input}`)
+    }
+  },
+})
+
+/**
+ * Validates that a string does not contain any invalid path characters.
+ * Defaults to the current working directory.
+ */
+export const path = build({
+  parse: async input => {
+    const invalidPathChars = /[["?<>|\]]+/
```

Comment:
`\` is a valid directory separator for Windows

Reply:
You are reading that wrong. It is escaping `]` so it doesn't close the group.
```
> invalidPathChars.test('someDir]')
true // invalid
> invalidPathChars.test('someDir\dsadf')
false // valid
> invalidPathChars.test('someDir\\dsadf')
false // valid
```

Reply:
Ooooohhhh, duh 🤦🏼 
Thanks.